src/parallel/GeneralDomainDecomposition.cpp

```diff
@@ -235,6 +249,54 @@ void GeneralDomainDecomposition::readXML(XMLfileUnits& xmlconfig) {
 	xmlconfig.getNodeValue("initialPhaseFrequency", _initFrequency);
 	global_log->info() << "GeneralDomainDecomposition frequency for initial rebalancing phase: " << _initFrequency
 					   << endl;
+
+	std::string gridSizeString;
+	if (xmlconfig.getNodeValue("gridSize", gridSizeString)) {
+		global_log->info() << "GeneralDomainDecomposition grid size: " << gridSizeString << endl;
+
+		if (gridSizeString.find(',') != std::string::npos) {
+			auto strings = string_utils::split(gridSizeString, ',');
+			if (strings.size() != 3) {
+				global_log->error()
+					<< "GeneralDomainDecomposition's gridSize should have three entries if a list is given, but has "
+					<< strings.size() << "!" << endl;
+				Simulation::exit(8134);
+			}
+			_gridSize = {std::stod(strings[0]), std::stod(strings[1]), std::stod(strings[2])};
+		} else {
+			double gridSize = std::stod(gridSizeString);
+			_gridSize = {gridSize, gridSize, gridSize};
+		}
```

Comment:
feels like this should be converted to some utils function

Reply:
could be done, but here for now ;)